### PR TITLE
BCC compatibility + some older problem fixes.

### PR DESCRIPTION
**Pull request by Road-block**
BCC compatibility is a side-effect from a bunch of dead code fixes.

Commit messages have the details.

### CODE REVIEW

**Comment by Road-block** [comment]
This supercedes #218 #214 completely, and #211 partially. It does not include the chat bubble border coloring fix.

**Comment by Road-block** [comment]
So.. I'm going to leave it alone for now.
With the last commits it can close #218 , #214 and #211 

I tried to keep things nice and separate in different commits rather than a big one so they can be cherry picked as needed.

**Comment by Road-block** [comment]
This should also close #219 #206  #202 and #221

**Comment by MeikTranel** [comment]
Try this in the meantime https://github.com/MeikTranel/prat-3-0/actions/runs/875606682

**Comment by Road-block** [comment]
Since there might be some non-devs landing here from Curseforge for who a PR is "useless".
You can test the PR as a full addon by getting this package.

[Prat-3.0-3.9.1-2-5df530f-bcc.zip](https://github.com/Road-block/prat-3-0/releases/download/3.9.1/Prat-3.0-3.9.1-2-5df530f-bcc.zip)

**Comment by Road-block** [comment]
New package for testing.

[Prat-3.0-3.9.1-5-571c1f6-bc.zip](https://github.com/Road-block/prat-3-0/releases/download/3.9.1-5/Prat-3.0-3.9.1-5-571c1f6-bc.zip)

Download, extract in AddOns, answer Yes to any overwrite prompts.

**Comment by Daeveren** [comment]
> New package for testing.
> 
> [Prat-3.0-3.9.1-3-085c8d7-bc.zip](https://github.com/Road-block/prat-3-0/releases/download/3.9.1/Prat-3.0-3.9.1-3-085c8d7-bc.zip)
> 
> Download, extract in AddOns, answer Yes to any overwrite prompts.

The scrolldown button no longer appears when I scroll chat to see old messages (display > buttons > show scrolldown reminder)

**Comment by Road-block** [comment]
> > New package for testing.
> > [Prat-3.0-3.9.1-3-085c8d7-bc.zip](https://github.com/Road-block/prat-3-0/releases/download/3.9.1/Prat-3.0-3.9.1-3-085c8d7-bc.zip)
> > Download, extract in AddOns, answer Yes to any overwrite prompts.
> 
> The scrolldown button no longer appears when I scroll chat to see old messages (display > buttons > show scrolldown reminder)

What version of the game (BCC, Classic Era)?  
Was it working with a previous Prat version? On what version of the game was it working if yes.  
Is it with OriginalButtons module enabled or not?

**Comment by ConKou** [comment]
Is there any way to make it so that WoWUp and other addon managers, can auto-detect files from PRs and update automatically? Because unless we do it manually, they still see Prat as a classic addon and issue a warning

**Comment by Road-block** [comment]
> Is there any way to make it so that WoWUp and other addon managers, can auto-detect files from PRs and update automatically? Because unless we do it manually, they still see Prat as a classic addon and issue a warning

There are workarounds for WowUp and Ajour (not sure about Overwolf client) to point it to a new repository for updates. 
I know some people close to Sylvanaar are trying to figure out a way to hand over the project so it can continue but that will take some time.

I could probably keep Prat going for Classic Era / BCC but I do not play retail, nor do I intend to in the future so I will not be involved in maintaining the addon going forward.
*It also has a huge install base and that's more "work" to support it than I'm willing to give*.

**TL;DR**: Wait for a new main repository to show up or the project to be handed over.
In the mean-time you *could* use WowUp's Install from URL and point it to this tag
https://github.com/Road-block/prat-3-0/releases/latest
wowup
 for BCC

**Comment by Daeveren** [comment]
> > > New package for testing.
> > > [Prat-3.0-3.9.1-3-085c8d7-bc.zip](https://github.com/Road-block/prat-3-0/releases/download/3.9.1/Prat-3.0-3.9.1-3-085c8d7-bc.zip)
> > > Download, extract in AddOns, answer Yes to any overwrite prompts.
> > 
> > 
> > The scrolldown button no longer appears when I scroll chat to see old messages (display > buttons > show scrolldown reminder)
> 
> What version of the game (BCC, Classic Era)?
> Was it working with a previous Prat version? On what version of the game was it working if yes.
> Is it with OriginalButtons module enabled or not?

In BCC it doesn't show the button. I only noticed this now installed your fork few days ago (before that I was using the normal addon, with some fixes from Curse comments).

**Comment by ConKou** [comment]
> > > > New package for testing.
> > > > [Prat-3.0-3.9.1-3-085c8d7-bc.zip](https://github.com/Road-block/prat-3-0/releases/download/3.9.1/Prat-3.0-3.9.1-3-085c8d7-bc.zip)
> > > > Download, extract in AddOns, answer Yes to any overwrite prompts.
> > > 
> > > 
> > > The scrolldown button no longer appears when I scroll chat to see old messages (display > buttons > show scrolldown reminder)
> > 
> > 
> > What version of the game (BCC, Classic Era)?
> > Was it working with a previous Prat version? On what version of the game was it working if yes.
> > Is it with OriginalButtons module enabled or not?
> 
> In BCC it doesn't show the button. I only noticed this now installed your fork few days ago (before that I was using the normal addon, with some fixes from Curse comments).

Its a thing with OriginalButton and Buttons Modules. Play around with them and you will find the fix. Happened to me as well, had to reload them one by one for it to work in the end

**Comment by Daeveren** [comment]
Only when I disabled Buttons and enabled Original Buttons I got the original up/down and Scrooldown buttons, but that's not what I wanted (and that disables the Buttons section in the Prat options completely). I didn't managed to make the Scrolldown button load from the Buttons module.
So far testing each of the buttons in the Buttons category goes like this:
- show arrows - works
- show social menu - does nothing
- show minimize button - does nothing
- show channel button - works
- show scrolldown reminder - does nothing
- show menu - works
- show voice buttons - does nothing.

**Comment by Road-block** [comment]
> Only when I disabled Buttons and enabled Original Buttons I got the original up/down and Scrooldown buttons, but that's not what I wanted (and that disables the Buttons section in the Prat options completely). I didn't managed to make the Scrolldown button load from the Buttons module.
> So far testing each of the buttons in the Buttons category goes like this:
> 
> * show arrows - works
> * show social menu - does nothing
> * show minimize button - does nothing
> * show channel button - works
> * show scrolldown reminder - does nothing
> * show menu - works
> * show voice buttons - does nothing.

I'll take a look when I find some time but if I'm honest I did not touch anything related to those modules (other than fixing a bug in OriginalButtons module) so I doubt this is a new bug.  
Probably default UI has changed and Prat has not followed those changes.

**Comment by ConKou** [comment]
Screenshot_5
small thing, in colored chat bubbles there is the little part where the bubble starts that doesnt get colored

**Comment by Road-block** [comment]
New package with minor fix for bubble coloring. [Prat-3.0-3.9.1-5-571c1f6-bc.zip](https://github.com/Road-block/prat-3-0/releases/download/3.9.1-5/Prat-3.0-3.9.1-5-571c1f6-bc.zip)

**Comment by watchout-mods** [comment]
@Road-block Oh nice, I didn't get a notification for this, bad github! I see you're much more invested in maintaining Prat :)

@KonstantinosKoubetsos I think the tail was never coloured in prat, or at least when I updated the code in my PR, it seemed so

**Comment by ConKou** [comment]
@Road-block thanks for the tail fix
Im noticing on WoW Interface its updated to the bcc libraries and picked up by WoW Up
https://www.wowinterface.com/downloads/info10783#other
Is that you with the same file commits as here?

**Comment by Road-block** [comment]
> @Road-block thanks for the tail fix
> Im noticing on WoW Interface its updated to the bcc libraries and picked up by WoW Up
> https://www.wowinterface.com/downloads/info10783#other
> Is that you with the same file commits as here?

It is.

**Comment by ConKou** [comment]
> > @Road-block thanks for the tail fix
> > Im noticing on WoW Interface its updated to the bcc libraries and picked up by WoW Up
> > https://www.wowinterface.com/downloads/info10783#other
> > Is that you with the same file commits as here?
> 
> It is.

That's great. So i can just change providers and monitor from WoW Interface

**Comment by ConKou** [comment]
@Road-block weird thing, i just installed from WoW Interface and colored bubbles stopped working all together. I have no color at all. I checked your commit and the lines of code are not existant on the file.

**Comment by Road-block** [comment]
> @Road-block weird thing, i just installed from WoW Interface and colored bubbles stopped working all together. I have no color at all. I checked your commit and the lines of code are not existant on the file.

I just downloaded manually to check I uploaded the correct package and I did.
It's possible your updater is picking the main addon rather than the patched one.
https://www.wowinterface.com/downloads/info26044

**Comment by ConKou** [comment]
Another minor thing i noticed. Colored bubbles dont work inside (certain?) instances. Confirmed by going in Blood Furnance. Inside theres no coloring, while outside there is

**Comment by Road-block** [comment]
> Another minor thing i noticed. Colored bubbles dont work inside (certain?) instances. Confirmed by going in Blood Furnance. Inside theres no coloring, while outside there is

That is likely a Blizzard limitation, same as for friendly Nameplates. Some frames cannot be modified when inside instances.

**Comment by ConKou** [comment]
I am getting the 'no player named xxx currently playing' spam in my chat pretty regularly and some Ace library lua errors when doing scan on the auction house with TSM. Since prat is the only not officially tbc supported addon i am running, im inclined to think theres a problem there, but i really can't tell for sure because the chat spam error comes and goes randomly.

**Comment by Road-block** [comment]
> I am getting the 'no player named xxx currently playing' spam in my chat pretty regularly and some Ace library lua errors when doing scan on the auction house with TSM. Since prat is the only not officially tbc supported addon i am running, im inclined to think theres a problem there, but i really can't tell for sure because the chat spam error comes and goes randomly.

https://github.com/sylvanaar/prat-3-0/issues **with** the actual errors posted. If actual Prat files (not libraries) are not mentioned anywhere in the callstack it means nothing.

**Comment by Road-block** [comment]
[Prat-3.0-3.9.1-5-571c1f6-bc.zip](https://github.com/Road-block/prat-3-0/releases/download/3.9.1-5/Prat-3.0-3.9.1-5-571c1f6-bc.zip) with fix for buttons module.

**Comment by Daeveren** [comment]
> [Prat-3.0-3.9.1-5-571c1f6-bc.zip](https://github.com/Road-block/prat-3-0/releases/download/3.9.1-5/Prat-3.0-3.9.1-5-571c1f6-bc.zip) with fix for buttons module.

I can confirm that the ScrollDown button now **works**!
Thank you!

**Comment by dfherr** [comment]
`GROUP_ROSTER_UPDATE` triggers on party and raid changes, but the `updateParty` method only checks for party units. `RAID_ROSTER_UPDATE` is also deprecated since MOP.

I just fixed it for myself like this:
in `onModuleEnable`:

```
- self:RegisterEvent("RAID_ROSTER_UPDATE", "updateRaid")
- if select(4, GetBuildInfo()) < 80000 and select(4, GetBuildInfo()) >= 20000 then
-     self:RegisterEvent("PARTY_MEMBERS_CHANGED", "updateParty")
- end
+self:RegisterEvent("GROUP_ROSTER_UPDATE", "updateGroup")
```
and right above the updateRaid/updateParty calls:
```
function module:updateGroup()
    if IsInRaid() then
      self:updateRaid()
    elseif IsInGroup() then
      self:updateParty()
    end
  end
  ```

probably not ideal, but a bit cleaner.

**Comment by Road-block** [comment]
New package with some cleanup of dead code per dfherr comment

[Prat-3.9.1-6-f65d85e-bc.zip](https://github.com/Road-block/prat-3-0/releases/download/3.9.1-6/Prat-3.9.1-6-f65d85e-bc.zip)

Also reminder that you can feed your WowUp or Ajour addon updater `Install from URL` with
[https://github.com/Road-block/prat-3-0/releases/latest](https://github.com/Road-block/prat-3-0/releases/latest) link
to get updates like this automatically.
